index.js

```diff
@@ -46,9 +46,18 @@ VirtualModulesPlugin.prototype.writeModule = function(filePath, contents) {
 
   debug(self._compiler.name, "Write module:", modulePath, contents);
 
+  // When using the WatchIgnorePlugin (https://github.com/webpack/webpack/blob/52184b897f40c75560b3630e43ca642fcac7e2cf/lib/WatchIgnorePlugin.js),
+  // the original watchFileSystem is stored in `wfs`. The following "unwraps" the ignoring
+  // wrappers, giving us access to the "real" watchFileSystem.
+  let finalWatchFileSystem = self._watcher && self._watcher.watchFileSystem;
+
+  while (finalWatchFileSystem && finalWatchFileSystem.wfs) {
```

Comment:
Thanks, the general approach looks good to me. I'm wonder however, why do we need while loop here?

Reply:
When you use multiple `WatchIgnorePlugin`, each of them overwrites the `watchFileSystem`, using the previous `watchFileSystem` as `wfs`: https://github.com/webpack/webpack/blob/52184b897f40c75560b3630e43ca642fcac7e2cf/lib/WatchIgnorePlugin.js#L98-L101

Reply:
Aha, I see, thank you for clarification